scripts/globals/events/vanaversary.lua

```diff
@@ -0,0 +1,488 @@
+-------------------------------------
+-- Vana'versary Celebration Campaign
+-------------------------------------
+require("scripts/globals/items")
+require("scripts/globals/npc_util")
+require("scripts/globals/zone")
+-------------------------------------
+
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.vanaversary = xi.events.vanaversary or {}
+xi.events.vanaversary.npcData = xi.events.vanaversary.npcData or {}
+xi.events.vanaversary.entities = xi.events.vanaversary.entities or {}
+
+-------------------------------
+-- Define Event and Date to run
+-------------------------------
+
+local event = SeasonalEvent:new("VanaversaryCelebration")
+
+xi.events.vanaversary.enabledCheck = function()
+    local enabled = xi.settings.main.VANAVERSARY_ENABLED
+    local always  = xi.settings.main.VANAVERSARY_ALWAYS
+    local sMonth  = xi.settings.main.VANAVERSARY_START_MONTH
+    local sDay    = xi.settings.main.VANAVERSARY_START_DAY
+    local eMonth  = xi.settings.main.VANAVERSARY_END_MONTH
+    local eDay    = xi.settings.main.VANAVERSARY_END_DAY
+    local month   = tonumber(os.date("%m"))
+    local day     = tonumber(os.date("%d"))
+
+    if enabled then
+        if month >= sMonth and day >= sDay and month <= eMonth and day <= eDay then
+            return true
+        elseif always then
+            return true
+        end
+    else
+        return false
+    end
+end
+
+event:setEnableCheck(xi.events.vanaversary.enabledCheck)
+
+---------------------------------------
+-- If Vana'versary is active, show NPCs
+---------------------------------------
+
+xi.events.vanaversary.showEntities = function(enabled)
+    if enabled and #xi.events.vanaversary.entities == 0 then
+        xi.events.vanaversary.generateEntities()
+    end
+
+    for _, entityID in pairs(xi.events.vanaversary.entities) do
+        local entity = GetNPCByID(entityID)
+        if entity then
+            if enabled then
+                entity:setStatus(xi.status.NORMAL)
+            else
+                -- TODO: Why doesn't DISAPPEAR work here?
```

Comment:
My guess is the answer is packets. I think we broke something somewhere. Retain comment here till we fix it because it really should be what you tried to use 1st.